src/faction_camp.cpp

```diff
@@ -1420,20 +1408,136 @@ void basecamp::craft_construction( npc &p, const std::string &cur_id, const std:
+        return itm.is_seed() && itm.typeId() != "marloss_seed" && itm.typeId() != "fungal_seeds";
+    };
+
+    std::vector<std::string> plant_names;
+
+    //bay_json is what the are should look like according to jsons
+    tinymap bay_json;
+    bay_json.generate( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, calendar::turn );
+    //bay is what the area actually looks like
+    tinymap bay;
+    bay.load( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, false );
+    tripoint mapmin = tripoint( 0, 0, omt_tgt.z );
+    tripoint mapmax = tripoint( 2 * SEEX - 1, 2 * SEEY - 1, omt_tgt.z );
+    for( const tripoint &pos : bay.points_in_rectangle( mapmin, mapmax ) ) {
+        //Needs to be plowed to match json
+        if( is_dirtmound( pos, bay_json, bay ) && is_unplowed( pos, bay ) ) {
+            plots_plow += 1;
+        }
+        if( is_dirtmound( pos, bay, bay ) ) {
+            plots_empty++;
+        }
+        if( bay.furn( pos ) == furn_str_id( "f_plant_harvest" ) && !bay.i_at( pos ).empty() ) {
+            const item &seed = bay.i_at( pos ).front();
+            if( is_valid_seed( seed ) ) {
+                harvestable++;
+                const islot_seed &seed_data = *seed.type->seed;
+                item tmp = item( seed_data.fruit_id, calendar::turn );
+                if( std::find( plant_names.begin(), plant_names.end(),
+                               tmp.type_name( 3 ) ) != plant_names.end() ) {
+                    plant_names.push_back( tmp.type_name( 3 ) );
+                }
+            }
         }
     }
 
+    int total_c = 0;
+    for( const std::string &i : plant_names ) {
+        if( total_c < 5 ) {
+            crops += "\t" + i + " \n";
+            total_c++;
+        } else if( total_c == 5 ) {
+            crops += "+ more \n";
+            break;
+        }
+    }
+
+    if( plow ) {
+        return plots_plow;
+    } else if( plant ) {
+        return plots_empty;
+    } else if( harvest ) {
+        return harvestable;
+    }
+    return 0;
+}
+
+void talk_function::camp_farm_start( npc &p, const std::string &dir, const tripoint &omt_tgt,
+                                     farm_ops op )
+{
+    bool plow = op & farm_ops::plow;
+    bool plant = op & farm_ops::plant;
+    bool harvest = op & farm_ops::harvest;
+    std::string crops_null;
+    size_t plots_cnt = camp_farm_act( omt_tgt, crops_null, op );
+    time_duration work = 0_minutes;
+    npc_ptr comp = nullptr;
+    if( harvest && plots_cnt ) {
+        work += 3_minutes * plots_cnt;
+        comp = individual_mission( p, _( "begins to harvest the field..." ),
+                                   "_faction_exp_harvest_" + dir, false, {}, "survival" );
+    }
+    if( plant && plots_cnt ) {
+        inventory total_inv = g->u.crafting_inventory();
+        std::vector<item *> seed_inv = total_inv.items_with( []( const item & itm ) {
+            return itm.is_seed() && itm.typeId() != "marloss_seed" && itm.typeId() != "fungal_seeds";
```

Comment:
This condition is repeated below again, it should probably be a static function instead.